command/agent/consul/connect.go

```diff
@@ -120,7 +117,25 @@ func connectSidecarRegistration(serviceId string, css *structs.ConsulSidecarServ
 				Name:         "Connect Sidecar Aliasing " + serviceId,
 				AliasService: serviceId,
 			},
-		},
+		}
+	} else {
+		checks = make([]*api.AgentServiceCheck, len(css.Checks))
+		for i, c := range css.Checks {
+			check, err := createCheckReg(serviceId, "", c, cMapping.HostIP, cMapping.Value, "")
```

Comment:
Not sure if I'm doing this correctly, and what host/port should be used.  When I tried, I can successfully set a custom `TCP` check; but cannot seem to get an `HTTP` to pass. The default envoyproxy seems to reject HTTP connection?!